include/swift/ABI/MetadataValues.h

```diff
@@ -361,10 +361,14 @@ enum class TypeReferenceKind : unsigned {
   /// unused.
   IndirectObjCClass = 0x03,
 
+  /// The conformance is for a non-nominal type referenced directly;
+  /// getTypeMetadata() points to the type metadata.
+  DirectTypeMetadata = 0x04,
+
   // We only reserve three bits for this in the various places we store it.
 
   First_Kind = DirectTypeDescriptor,
-  Last_Kind = IndirectObjCClass,
+  Last_Kind = DirectTypeMetadata,
```

Comment:
If you were going to generalize this to all tuples, you might have this new type reference kind instead be a reference to a mangled name in the generic context of the conformance. That should allow any kind of type to be instantiated without hardcoding type-specific metadata access patterns.

Reply:
Yeah.  If we're preparing for arbitrary structural conformances in this PR, there's no sense in adding the ability to directly link to metadata when that's only likely to ever be useful for the  empty tuple.  This could link to a mangled name, although that might unnecessarily duplicate the information in the conformance; alternatively, it could "link" to a `MetadataKind`.